Blog post for Kroxylicious v0.10.0 (#96)
kroxylicious/kroxylicious.github.io: _posts/2025-01-30-release-0_10_0.md

---
layout: post
title:  "Kroxylicious release 0.10.0"
date:   2025-01-30 00:00:00 +0000
author: "Grace Grimwood"
author_url: "https://github.com/gracegrimwood"
# noinspection YAMLSchemaValidation
categories: blog kroxylicious releases
tags: [ "releases", "kroxylicious" ]
---

The Kroxylicious project is very pleased to announce the release of [Kroxylicious 0.10.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.10.0). See the [Changelog](https://github.com/kroxylicious/kroxylicious/blob/main/CHANGELOG.md#0100) for a list of changes and summary of Deprecations, Changes and Removals.

> ### Fortanix KMS for Record Encryption

Reviewer: Fortanix is the company.   They have several products.  Fortanix DSM is one of the them.  It is Fortanix DSM that we have the plugin built for.

```suggestion
### Fortanix DSM KMS for Record Encryption
```

Reviewer: We should also acknowledge Fortanix's trademark.

> This release introduces initial support for using [Fortanix Data Security Manager](https://www.fortanix.com/platform/data-security-manager) as a Key Management System provider with the Kroxylicious Record Encryption filter ([#1672](https://github.com/kroxylicious/kroxylicious/pull/1672)).

Reviewer: "within" reads better to me.

```suggestion
This release introduces initial support for using [Fortanix Data Security Manager](https://www.fortanix.com/platform/data-security-manager) as a Key Management System provider within the Kroxylicious Record Encryption filter ([#1672](https://github.com/kroxylicious/kroxylicious/pull/1672)).
```

### Improvements and tweaks to Record Encryption

> We've introduced some changes in this release to Encryption DEK invalidation logic to improve coordination between threads ([#1643](https://github.com/kroxylicious/kroxylicious/pull/1643)), and added some restrictions to TLS protocols and cipher suites used by Key Management System implementations to improve the overall security of Kroxylicious ([#1743](https://github.com/kroxylicious/kroxylicious/pull/1743)).

Reviewer 2: The cipher suite and protocol restrictions also apply to downstream connections not just the KMS

Reviewer: upstream and downstream kafka connections.

Reviewer 3: `added some restrictions` sounds a bit like we've changed some default behaviour. Is this better described as "enabling users to control which TLS protocols and cipher suites are enabled for ..."

> We've introduced some changes in this release to Encryption DEK invalidation logic to improve coordination between threads ([#1643](https://github.com/kroxylicious/kroxylicious/pull/1643)), and added some restrictions to TLS protocols and cipher suites used by Key Management System implementations to improve the overall security of Kroxylicious ([#1743](https://github.com/kroxylicious/kroxylicious/pull/1743)).

Reviewer: ```suggestion
We've introduced some changes in this release to Encryption DEK invalidation logic to improve coordination between threads ([#1643](https://github.com/kroxylicious/kroxylicious/pull/1643)), and added the ability to restrict which TLS protocols/cipher suites are available for Kafka connections (upstream and downstream) as well as the TLS made to KMS providers ([#1743](https://github.com/kroxylicious/kroxylicious/pull/1743).
```

### More configurability for SNI

> This release introduces the `advertisedBrokerAddressPattern` property for SNI routing proxy configurations, allowing greater customisation for proxies using SNI ([#1761](https://github.com/kroxylicious/kroxylicious/pull/1761)).

Reviewer 2: Should we expand one of the usages SNI to Server Name Indicator?

### Feedback

> Please let us know, through [Slack](https://kroxylicious.slack.com), [GitHub](https://github.com/kroxylicious/kroxylicious/issues) or even [bsky](https://bsky.app/profile/kroxylicious.io)), if you find the project interesting or helpful and especially if you have deployed it in production.

Reviewer: I think we should call out [Alan Robinson](https://github.com/alanrobinson-dwp) as a contributor.

Author: Done!
